adaguc-checker.py

```diff
@@ -57,98 +68,163 @@ def checker(self, filename):
 
     def getmap(self, source, layer):
-        #print "Obtaining report and data for layer", layer
+        """
+        Performs a get map request for the specified layer. This results in an image of the getmap result as well as
+        a checker report written to the output directory of the checker.
+        :param source: autoWMS directory where the file to check is located
+        :param layer: layer dictionary including the name and the boundingbox of the layer
+        :return: image data for the get map result.
+        """
+
+        layer_par = '='.join(("LAYERS", layer["name"]))
+
+        # Note: The order of the boundingbox definition is different for the baselayer service and ADAGUC autoWMS
+        bounding_box_arg = ','.join((layer["bbox"]["minx"], layer["bbox"]["miny"], layer["bbox"]["maxx"], layer["bbox"]["maxy"]))
+        bounding_box_par = '='.join(("BBOX", bounding_box_arg))
+
+        get_map_request = ''.join((self.base_url, '&'.join((source, layer_par, query_string_map, query_string_par_layer, bounding_box_par))))
+        logger.debug("get_map_request:\n" + get_map_request)
 
-        layer_par = '='.join(("LAYERS", layer))
-        get_map_request = ''.join((self.base_url, '&'.join((source, layer_par, query_string_map, query_string_par))))
-        #print "URL:", get_map_request
         try:
             with closing(urlopen(url=get_map_request, context=ssl._create_unverified_context())) as r:
                 imgdata = r.read()
             if self.imagedir and os.path.exists(self.imagedir):
-                imgfile = open("%s/%s.%s.png" % (self.imagedir, self.fname, layer), "wb")
+                imgfile = open("%s/%s.%s.png" % (self.imagedir, self.fname, layer["name"]), "wb")
                 imgfile.write(imgdata)
                 imgfile.close()
             elif not os.path.exists(self.imagedir):
                 print >>sys.stderr, "%s doesn't exists. Not writing image file."
-                
         except: pass
 
-        #print ("========= BEGIN GETMAP REPORT ==========")
+        return imgdata
+
+    def getbaselayers(self, layer_bbox):
+        """
+        Retrieves a background layer and country outlines for the given bounding box from geoservices.knmi.nl.
+        :param layer_bbox: The required bounding box.
+        :return: Image data for the background layer and the country outlines
+        """
+
+        # Note: The order of the boundingbox definition is different for the baselayer service and ADAGUC autoWMS
+        bounding_box_arg = ','.join((layer_bbox["miny"], layer_bbox["minx"], layer_bbox["maxy"], layer_bbox["maxx"]))
+        bounding_box_par = '='.join(("BBOX", bounding_box_arg))
+
+        # Retrieve the background map.
+        get_bgmap_request = ''.join((base_url_bgmap, '&'.join((
+            "LAYERS=naturalearth2", query_string_map, query_string_par_baselayer, bounding_box_par))))
+        logger.debug("get_bgmap_request:\n" + get_bgmap_request)
```

Comment:
Is het de bedoeling dat de inhoud van get_bgmap_request op een aparte regel gelogd wordt?

Reply:
Besproken met Saskia: dit is inderdaad de bedoeling.